src/platform/qt/ScriptingTextBufferModel.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2013-2022 Jeffrey Pfau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#pragma once
+
+#include <QAbstractListModel>
+
+class mScriptTextBuffer;
+
+namespace QGBA {
+
+class ScriptingTextBuffer;
+
+class ScriptingTextBufferModel : public QAbstractListModel {
+Q_OBJECT
+friend class ScriptingController;
+
+public:
+	enum ItemDataRole {
+		DocumentRole = Qt::UserRole + 1,
+	};
+
+	ScriptingTextBufferModel(QObject* parent = nullptr);
+
+	int rowCount(const QModelIndex& parent = QModelIndex()) const;
+	QVariant data(const QModelIndex& index, int role = Qt::DisplayRole) const;
+
+signals:
+	void textBufferCreated(ScriptingTextBuffer*);
+
+public slots:
+	void reset();
+
+private slots:
+	void bufferNameChanged(const QString&);
+
+private:
+	static mScriptTextBuffer* createTextBuffer(void* context);
```

Comment:
I think I'd prefer if this is public instead of using friend classes. Since this isn't a library, we don't have to worry as much about API misuse.

Reply:
Picked a slightly different solution that should still satisfy.